akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/adapter/GuardianStartupSpec.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.scaladsl.adapter
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+import akka.actor.ActorSystemImpl
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import akka.actor.testkit.typed.scaladsl.TestProbe
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.scaladsl.Behaviors
+import org.scalatest.Matchers
+import org.scalatest.WordSpec
+import org.scalatest.concurrent.ScalaFutures
+
+class GuardianStartupSpec extends WordSpec with Matchers with ScalaFutures {
+
+  "The user guardian" must {
+
+    "should get a message sent to it early" in {
+      var system: ActorSystem[String] = null
+      val sawMsg = new CountDownLatch(1)
+      val guardianBehavior = Behaviors.receiveMessage[String] { msg =>
+        if (msg == "msg") sawMsg.countDown()
+        Behaviors.same
+      }
+      try {
+        system = ActorSystem(guardianBehavior, "GuardianStartupSpec-get-all")
+        system ! "msg"
+
+        sawMsg.await(3, TimeUnit.SECONDS) should ===(true)
+
+      } finally {
+        if (system ne null)
+          ActorTestKit.shutdown(system)
```

Comment:
Not that I like a sharable `private var system: ActorSystem[_]` but all of these could reduce boilerplate of each declaration of null, and all the `finally`s with an `afterEach` setting to null .
```scala
if (system ne null) ActorTestKit.shutdown(system)
system = null
```

Reply:
I tried making a setup trait but it basically gives the same amount of code in each test and is much harder to reason about. I think the (little) boilerplate is ok here.